CHANGELOG.md

```diff
@@ -1,5 +1,116 @@
 # Changelog
 
+## 🏄‍♀️ 1.1.1
+
+- ### Features
+
+  - **Improve JSON errors debuggability - [xtuc], [pull/394]**
+
+    This PR improves JSON error output in `wrangler`. Specifically:
+
+    - If a `package.json` file fails to decode, `wrangler` now emits a clearer error:
+
+      ```
+      $ wrangler build
+      ⬇️  Installing wranglerjs...
+      ⬇️  Installing wasm-pack...
+      thread 'main' panicked at 'could not parse "./package.json": Error("expected `,` or `}`", line: 4, column: 3)', src/libcore/result.rs:999:5
+      ```
+
+    - If the `wranglerjs` backend returns invalid JSON, it now preserves the output file for further investigation. Note that the console doesn't print the output file location by default, and you will need to pass `RUST_LOG=info` while running `wrangler build`, and search for the `--output-file=FILE` argument passed to `wranglerjs`:
+
+      ```
+      $ RUST_LOG=info wrangler build
+      ⬇️ Installing wasm-pack...
+      [2019-08-09T19:28:48Z INFO  wrangler::commands::build::wranglerjs] Running "/Users/kristian/.nvm/versions/node/v12.1.0/bin/node" "/Users/kristian/src/workers/wrangler/wranglerjs" "--output-file=/var/folders/5x/yzqyqst11n518yl8xl7yv1f80000gp/T/.wranglerjs_output5eREv" # ...
+      ```
+
+    - If the preview service returns invalid JSON, it now emits a clearer error, and the full output can be seen by using `RUST_LOG=info`.
+
+      Previously:
+
+        ```
+        $ wrangler preview
+        ⬇️ Installing wasm-pack...
+        ⬇️ Installing wranglerjs...
+        ✨   Built successfully.
+        Error: Error("expected value", line: 2, column: 1)
+        ```
+
+      Now:
+
+      ```
+      $ wrangler preview
+      ⬇️ Installing wranglerjs...
+      ⬇️ Installing wasm-pack...
+      ✨   Built successfully, built project size is 1 MiB. ⚠️  Your built project has grown past the 1MiB size limit and may fail to deploy. ⚠️  ✨
+      Error: https://cloudflareworkers.com/script: Server Error: 502 Bad Gateway
+      ```
+
+    [xtuc]: https://github.com/xtuc
+    [pull/394]: https://github.com/cloudflare/wrangler/pull/394
+
+- ### Fixes
+
+  - **Fix `wrangler config` for systems with non-unix EOL - [xortive], [issue/389][pull/399]**
+
+    `wrangler config` was not properly truncating whitespace from the end of user input, resulting in a panic when trying to use `wrangler publish`, because `wrangler` would try to create an HTTP header with invalid characters. Now, `wrangler` will properly truncate extra whitespace (including `\r`) from the end of the input into `wrangler config`
```

Comment:
```suggestion
    `wrangler config` was not properly truncating whitespace from the end of user input, resulting in a panic when trying to use `wrangler publish`, because `wrangler` would try to create an HTTP header with invalid characters. Now, `wrangler` will properly truncate extra whitespace (including `\r`) from the end of the input into `wrangler config`.
```